### PR TITLE
added NotNil query part

### PR DESCRIPTION
**Pull request by woutslakhorst**
required for checking if a specific claim is available in a verifiable credential. Eg: any VC with an organization name.

### CODE REVIEW

File: index_test.go [hunk 1]
```diff
@@ -349,6 +349,23 @@ func TestIndex_Find(t *testing.T) {
...
+			b := testBucket(t, tx)
+			return i.Iterate(b, q, func(key Reference, value []byte) error {
+				count++
+				return nil
+			})
+		})
+
+		assert.NoError(t, err)
+		// it's a triple index where 4 matching trees exist
+		assert.Equal(t, 4, count)
```
Comment (reinkrul) [inline index_test.go:366]: I don't understand this, in `jsonExample` and `jsonExample2` there's summed up only 2 `part` fields?

And there's no non-matching document in the store? (testing `NotNil` doesn't match nil fields?)

File: search.go [hunk 2]
```diff
@@ -128,6 +128,14 @@ func Range(queryPath QueryPath, begin Scalar, end Scalar) QueryPart {
 	}
 }
 
+// NotNil creates a query part where the value must exist.
+// This is done by finding results between byte 0x0 and 0xff
```
Comment (reinkrul) [inline search.go:132]: `0xff` is actually just the last entry in the bucket? Since the keys are not 1-byte long? (well the might be, but probably not?)

Comment (woutslakhorst) [reply search.go:132]: This is how the cursor works. It uses bytes.Compare so any value where the first byte is smaller than 0xff will match.

File: index_test.go (hunk 1 above)
Comment (woutslakhorst) [reply index_test.go:366]: undoubling is done at a different level (QueryPlan collector)

**Review by reinkrul — APPROVED** [review]

**Pull request merged**